Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -2555,7 +2597,14 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
     void vectorAvgRound(SIMDInfo simdInfo, FPRegisterID a, FPRegisterID b, FPRegisterID dest) { UNUSED_PARAM(simdInfo); UNUSED_PARAM(a); UNUSED_PARAM(b); UNUSED_PARAM(dest); }
     void vectorMulSat(FPRegisterID a, FPRegisterID b, FPRegisterID dest) { UNUSED_PARAM(a); UNUSED_PARAM(b); UNUSED_PARAM(dest); }
     void vectorDotProductInt32(FPRegisterID a, FPRegisterID b, FPRegisterID dest, FPRegisterID) { UNUSED_PARAM(a); UNUSED_PARAM(b); UNUSED_PARAM(dest); }
-    void vectorSwizzle(FPRegisterID a, FPRegisterID b, FPRegisterID dest) { UNUSED_PARAM(a); UNUSED_PARAM(b); UNUSED_PARAM(dest); }
+
+    void vectorSwizzle(FPRegisterID a, FPRegisterID b, FPRegisterID dest)
+    {
+        if (a != dest)
+            m_assembler.movapd_rr(a, dest);
+        m_assembler.pshufb(b, dest);
```

Comment:
This is SSE3 only, and x64 does not ensure SSE3 availability. Use check and have fallback.